src/sage/rings/polynomial/infinite_polynomial_element.py

```diff
@@ -1018,42 +1117,42 @@ def coefficient(self, monomial):
             2
 
         """
+        P = self.parent()
         if self._p == 0:
-            res = 0
-        elif isinstance(monomial, self.__class__):
-            if not (self.parent().has_coerce_map_from(monomial.parent())):
-                res = 0
+            return P.zero()
+        if isinstance(monomial, self.__class__):
+            if not (P.has_coerce_map_from(monomial.parent())):
```

Comment:
```suggestion
            if not P.has_coerce_map_from(monomial.parent()):
```
https://xkcd.com/297/ :)

Reply:
that was not me!  (and lisp always has the function first)!  :-)